src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -2088,7 +2088,10 @@ private void ReplaceAllInChunk(int[] replacements, int replacementsCount, String
                 fixed (char* valuePtr = value)
                 {
                     // calculate the total amount of extra space or space needed for all the replacements.
-                    int delta = (value.Length - removeCount) * replacementsCount;
+                    long longDelta = (value.Length - removeCount) * (long)replacementsCount;
+                    int delta = (int)longDelta;
+                    if (delta != longDelta)
```

Comment:
I could have used this trick here (to avoid exception)? https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/System/Environment.Unix.cs#L352

Reply:
yup